### PR TITLE
[MODTEMPENG-34] Provide two options for barcode tokens on patron notices 

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**

### CODE REVIEW

File: ramls/template-engine.raml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 title: Template engine
 baseUri: http://api.example.com/{version}
-version: v2.1
+version: v2.2
```
Comment (kostyantyn-kh) [inline ramls/template-engine.raml]: Please change interface version in descriptor as well

File: src/main/java/org/folio/template/util/BarcodeImageGenerator.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Base64;
+
+public class BarcodeImageGenerator {
+  private static final Logger LOG = LoggerFactory.getLogger("mod-template-engine");
+  private static final String MIME_TYPE_PNG = "image/x-png";
```
Comment (kostyantyn-kh) [inline src/main/java/org/folio/template/util/BarcodeImageGenerator.java]: Is it standard mime type?
According to https://www.iana.org/assignments/media-types/media-types.xhtml#image
it's image/png

image/png
--




image/png
--




image/png
--

Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/template/util/BarcodeImageGenerator.java]: They are the same thing. And `Barcode4J`, the library I used to generate images, supports both. PNG used to be a non-standard MIME type up until 1996, hence the `x-` prefix :)

Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/template/util/BarcodeImageGenerator.java]: fixed

File: ramls/template-engine.raml (hunk 1 above)
Comment (OleksandrVidinieiev) [reply ramls/template-engine.raml]: fixed

File: src/test/java/org/folio/rest/impl/TemplateRequestTest.java [hunk 3]
```diff
@@ -560,11 +562,150 @@ public void shouldEnrichContextAndFormatDatesCorrectly() {
...
+        .post(TEMPLATE_REQUEST_PATH)
+        .then()
+        .statusCode(HttpStatus.SC_OK)
+        .body("templateId", is(templateId))
+        .body("result.header", is(expectedHeader))
+        .body("result.body", is(expectedBody))
+        .body("meta.lang", is(EN_LANG))
+        .body("meta.outputFormat", is(HTML_OUTPUT_FORMAT))
+        .body("result.attachments.size()", is(1));
   }
```
Comment (kostyantyn-kh) [inline src/test/java/org/folio/rest/impl/TemplateRequestTest.java]: Please create test that to verify case when template contains image tag other than barcodeImage: userImage, itemImage etc. Processor should not create attachment.

Comment (OleksandrVidinieiev) [reply src/test/java/org/folio/rest/impl/TemplateRequestTest.java]: done

**Pull request merged**